### DISCUSSION TITLE
SOE Tags for Crawlers to Index

### DISCUSSION THREAD

**Discussion by Overcon** [poblabs/weewx-belchertown#771] [Q&A]
Is there a section, or a means of adding SEO tags to the Belcher Skin to get our site brought up on web searches?

**Comment by roe-dl** [comment]
I did it using the new internationalization features of WeeWX 4.6. In the `Texts` section of the language configuration file I set a section for each page. In `header.html.tmpl` I replaced `#set $head_desc = $obs.label.html_description` by `#set $head_desc = $pgettext($page,'html_description')` and `#set $head_title = $obs.label.html_title` by `#set $head_title = $pgettext($page,'html_title')`

in `lang/<language-code>.conf`:
```
...
[Texts]
    [[home]]
        html_title = ...
        html_description = ...
    [[graphs]]
        html_title = ...
        html_description = ...
    [[records]]
        html_title = ...
        html_description = ...
    ...
```

**Reply by Overcon** [reply]
Do you have a write up on what to change or would you mind sharing that? I am not a programmer, but I can modify it to fit my site. I don't even know what much of that is other than in general terms. If not, that's OK. 

Thanks.

**Comment by roe-dl** [comment]
There is no programming. There is adjusting the skin only.

So, Basics first: Form version 4.6 on WeeWX includes a new internationalization feature. That is not only useful for internationalization but can also be used to create page dependent SEO information. 

To use it, you need a `lang` subdirectory within the Belchertown skin directory. Within that subdirectory you create a file named according to the code of your language. For example, if you speak german, the file name would be `de.conf`, if you speak french, the file name would `fr.conf` etc. pp. In `weewx.conf` you include a line `lang = your_language_code` in the Belchertown skin section, that is `lang = de` for the first example, and `lang = fr` for the second.

In general, that file looks like the `skin.conf` file, but it includes an additional section called `Texts`. In that section you write a subsection for each page of the skin as shown in my previous post. Each of that sections need the keys `html_title` and `html_description`, defining the appropriate text for the page.

Then you change the file `header.html.tmpl`. You replace 2 lines as described above.

That is all you need to have a different title and description for each page as required by the crawlers.

Please note, you need at least WeeWX 4.6 for that to work.

**Reply by Overcon** [reply]
Ok, thanks. I am going to dig into that now. I upgraded to 4.7, so should be good to go on the Weewx version. I am not sure what version of the Belchertown skin, but it doesn't sound like that matters.

Thanks!
